plutus-pab/src/Cardano/Protocol/Socket/Client.hs

```diff
@@ -33,52 +33,56 @@ import           Ouroboros.Network.Socket
 import           Cardano.Protocol.Socket.Type
 import           Ledger                                              (Block, Slot (..), Tx (..))
 
--- | Client handler, used to communicate with the client thread.
-data ClientHandler = ClientHandler
-    { chInputQueue  :: TQueue Tx
-    , chSocketPath  :: FilePath
-    , chCurrentSlot :: IO Slot
-    , chHandler     :: Block -> Slot -> IO ()
+data ChainSyncHandle = ChainSyncHandle
+    { cshCurrentSlot :: IO Slot
+    , cshHandler     :: Block -> Slot -> IO ()
     }
 
+newtype TxSendHandle = TxSendHandle
+    { tshQueue :: TQueue Tx }
+
 -- | Queue a transaction to be sent to the server.
 queueTx ::
-    ClientHandler
+    TxSendHandle
  -> Tx
  -> IO ()
-queueTx ClientHandler { chInputQueue } tx =
-    atomically (writeTQueue chInputQueue tx)
+queueTx TxSendHandle { tshQueue } tx =
+    atomically (writeTQueue tshQueue tx)
 
-getCurrentSlot :: ClientHandler -> IO Slot
-getCurrentSlot = chCurrentSlot
+getCurrentSlot :: ChainSyncHandle -> IO Slot
+getCurrentSlot = cshCurrentSlot
 
-runClientNode :: FilePath
+-- | Run the chain sync protocol to get access to the current slot number.
+runChainSync' :: FilePath
               -> SlotConfig
-              -> (Block -> Slot -> IO ())
-              -> IO ClientHandler
-runClientNode socketPath slotConfig onNewBlock = do
-    inputQueue  <- newTQueueIO
-    let handle = ClientHandler { chInputQueue = inputQueue
-                               , chSocketPath = socketPath
-                               , chCurrentSlot = currentSlot slotConfig
-                               , chHandler = onNewBlock
-                               }
+              -> IO ChainSyncHandle
+runChainSync' socketPath slotConfig =
+  runChainSync socketPath slotConfig (\_ _ -> pure ())
+
+runChainSync :: FilePath
+             -> SlotConfig
+             -> (Block -> Slot -> IO ())
+             -> IO ChainSyncHandle
+runChainSync socketPath slotConfig onNewBlock = do
+    let handle = ChainSyncHandle { cshCurrentSlot = currentSlot slotConfig
```

Comment:
`currentSlot` doesn't need any information from the server so maybe it shouldn't be part of the `ChainSyncHandle`?

Reply:
It needs to know when a new slot has been added (when a new block has been received). The client keeps track of this information, and it is needed to support the `GetCurrentSlot` interpreter. I could write it as an application of the more generic handler mechanism, but since we always need it I went for the smaller code.

Do you think I should also split the `GetCurrentSlot` into a new client type?

Reply:
I think we decided to keep things as they are, for now?